src/Oscoin/Crypto/Hash/RealWorld.hs

```diff
@@ -111,6 +121,16 @@ instance FromJSON (Hash Crypto) where
         either fail pure
             . second Hash . decodeAtBase BaseN.Base58btc . encodeUtf8
 
+instance ToJSON (ShortHash Crypto) where
+    toJSON     = toJSON . T.pack . show
+    toEncoding = toEncoding . T.pack . show
+
+instance FromJSON (ShortHash Crypto) where
+    parseJSON = withText "ShortHash" $ \t ->
+        case parseShortHash t of
+            Nothing -> fail "invalid short hash"
```

Comment:
Shouldn't we fail by also showing the input text, or do you feel that's redundant? I can imagine how this might be useful if this failure is triggered deep down the bowels of the app, and gets logged without any context.

Reply:
6949e75

Reply:
Looks good 😉